libs/wingc/src/jsify/codemaker.rs

```diff
@@ -42,8 +131,47 @@ impl CodeMaker {
 	/// Emits multiple lines of code starting with the current indent.
 	pub fn add_code(&mut self, code: CodeMaker) {
 		assert_eq!(code.indent, 0, "Cannot add code with indent");
-		for (indent, line) in code.lines {
-			self.lines.push((indent + self.indent, line));
+		for mut line_data in code.lines {
+			line_data.indent += self.indent;
+			self.push_line(line_data);
+		}
+	}
+
+	/// Emits code starting on the current line, mutating it if needed.
+	/// Note this mutates the given code.
+	pub fn append<S: Into<CodeMaker>>(&mut self, line: S) {
+		let mut new_code: CodeMaker = line.into();
+		if new_code.is_empty() {
+			return;
+		}
+
+		if self.lines.is_empty() {
+			self.empty_line();
+		}
+
+		if let Some(last_line) = self.lines.last_mut() {
+			let mut first_new_line = new_code.lines.remove(0);
+			last_line.line.push_str(&first_new_line.line);
+
+			let last_original_mapping = last_line.mappings.last().cloned();
+			let add_another_map = !first_new_line.mappings.is_empty();
+
+			for (offset, span) in first_new_line.mappings.drain(..) {
+				last_line
+					.mappings
+					.push((last_line.line.len() - first_new_line.line.len() + offset, span));
+			}
+
+			if add_another_map {
+				if let Some(last_original_mapping) = last_original_mapping {
+					last_line.mappings.push((last_line.line.len(), last_original_mapping.1));
+				}
+			}
+		}
```

Comment:
This section looks important, can we add a few comments?

Reply:
commented and fixed a bug 👍